src/atheros/atherosd.c

```diff
@@ -80,214 +86,140 @@ struct vel_profile inverse_kinematics(int8_t v_x, int8_t v_y, int8_t w)
     return vels;
 }
 
-void kill_robot(state_t *state)
+void send_command(state_t *state, struct command cmd)
 {
-    int8_t buffer[MAX_BUFFER];
+    pthread_mutex_lock(&state->atmegafp_mutex);
 
-    buffer[0] = MAGIC >> 24 & 0xff;
-    buffer[1] = MAGIC >> 16 & 0xff;
-    buffer[2] = MAGIC >>  8 & 0xff;
-    buffer[3] = MAGIC >>  0 & 0xff;
+    for(unsigned long i = 0; i < sizeof magic_bytes; i++)
+        fputc(magic_bytes[i], state->atmegafp);
 
-    buffer[4] = 0;
+    for(int8_t i = 0; i < cmd.cmd_size; i++)
+        fputc(cmd.cmd[i], state->atmegafp);
 
-    pthread_mutex_lock(&state->atmegafp_mutex);
-    write(state->atmegafd, buffer, 5);
-    if(debug)
-        printf("[atherosd] Killed the robot\n");
     pthread_mutex_unlock(&state->atmegafp_mutex);
 }
 
-void kill_motors(state_t *state)
+struct command kill_robot()
```

Comment:
all of these functions are just constructing specific command types right? Could we rename them so it doesn't sound like they are actually sending the commands? (i.e. rename `kill_robot()` to something like `gen_kill_robot_cmd()`)